external/fv3fit/fv3fit/train.py

```diff
@@ -55,6 +56,78 @@ def dump_dataclass(obj, yaml_filename):
         yaml.safe_dump(dataclasses.asdict(obj), f)
 
 
+def get_data(
+    training_data_config: str,
+    validation_data_config: Optional[str],
+    local_download_path: Optional[str],
+    variable_names: Sequence[str],
+) -> Tuple[loaders.typing.Batches, loaders.typing.Batches]:
+    if local_download_path is None:
+        return get_uncached_data(
+            training_data_config=training_data_config,
+            validation_data_config=validation_data_config,
+            variable_names=variable_names,
+        )
+    else:
+        return get_cached_data(
+            training_data_config=training_data_config,
+            validation_data_config=validation_data_config,
+            local_download_path=local_download_path,
+            variable_names=variable_names,
+        )
+
+
+def get_uncached_data(
+    training_data_config: str,
+    validation_data_config: Optional[str],
+    variable_names: Sequence[str],
+) -> Tuple[loaders.typing.Batches, loaders.typing.Batches]:
+    with open(training_data_config, "r") as f:
+        config = yaml.safe_load(f)
+    loader = loaders.BatchesLoader.from_dict(config)
+    logger.info("configuration loaded, creating batches object")
+    train_batches = loader.load_batches(variables=variable_names)
+    if validation_data_config is not None:
+        with open(training_data_config, "r") as f:
+            config = yaml.safe_load(f)
+        loader = loaders.BatchesLoader.from_dict(config)
+        logger.info("configuration loaded, creating batches object")
+        val_batches = loader.load_batches(variables=variable_names)
+    else:
+        val_batches = []
+    return train_batches, val_batches
+
+
+def get_cached_data(
+    training_data_config: str,
+    validation_data_config: Optional[str],
+    local_download_path: str,
+    variable_names: Sequence[str],
+) -> Tuple[loaders.typing.Batches, loaders.typing.Batches]:
+    train_data_path = os.path.join(local_download_path, "train_data")
+    logger.info("saving training data to %s", train_data_path)
+    os.makedirs(train_data_path, exist_ok=True)
+    save_main(
+        data_config=training_data_config,
+        output_path=train_data_path,
+        variable_names=variable_names,
+    )
+    train_batches = loaders.batches_from_netcdf(path=train_data_path)
+    if validation_data_config is not None:
+        validation_data_path = os.path.join(local_download_path, "validation_data")
+        logger.info("saving validation data to %s", validation_data_path)
+        os.makedirs(validation_data_path, exist_ok=True)
+        save_main(
+            data_config=validation_data_config,
+            output_path=validation_data_path,
+            variable_names=variable_names,
+        )
+        val_batches = loaders.batches_from_netcdf(path=train_data_path)
```

Comment:
The PR to add a variable names arg was merged, so `batches_from_netcdf` will need to be called with that arg.